### PR TITLE
feat: Add Setting Screen Menu Item Injection

### PR DESCRIPTION
**Pull request by sternetj**
## Changes

  - Adds the ability to modify default menu items and inject new ones via the developer config

## Screenshots

image
image

### CODE REVIEW

File: src/common/DeveloperConfig.ts [hunk 1]
```diff
@@ -112,6 +114,9 @@ export type DeveloperConfig = {
     educationContentOverrides?: Record<string, EducationContent>;
   };
   onAppSessionStart?: (params: OnAppSessionStartParams) => Promise<void>;
+  modifySettingScreenMenuItems?: (
+    items: { id: string; title: string; action: () => void }[],
+  ) => { id: string; title: string; action: () => void }[];
```
Comment (markdlv) [inline src/common/DeveloperConfig.ts:119]: So you get the default SDK menu items passed to you, then return the ones you want shown.  Beautiful.

File: src/hooks/useHttpClient.tsx [hunk 2]
```diff
@@ -48,7 +48,10 @@ export const HttpClientContextProvider = ({
   const { authResult, refreshForAuthFailure } = useAuth();
 
   const axiosInstance = injectedAxiosInstance || defaultAxiosInstance;
-  const apiClient = defaultAPIClientInstance;
+  const apiClient = useMemo(
+    () => new APIClient<RestAPIEndpoints>(axiosInstance),
+    [axiosInstance],
+  );
```
Comment (markdlv) [inline src/hooks/useHttpClient.tsx:54]: Why is this needed?

Comment (sternetj) [reply src/hooks/useHttpClient.tsx:54]: I wanted to provide an app config and it uses the new one query rest client that didn't work with the DataProviderDecorator prior to this change. I could probably back this out and just use the AppConfigContext.Provider instead.

**Review by markdlv — DISMISSED** [review]
See my note about useHttpClient file changes- seems unrelated

File: src/hooks/useHttpClient.tsx (hunk 2 above)
Comment (markdlv) [reply src/hooks/useHttpClient.tsx:54]: My gut reaction would be yes, if this is only beneficial for storyboard purposes, it'd be sweet to refactor to where this change isn't needed.  At the same time, I'm not overly concerned about the change, so if it's a can of worms nbd.

Comment (sternetj) [reply src/hooks/useHttpClient.tsx]: Not a big deal. I switched it to just providing my own config

File: src/common/DeveloperConfig.ts (hunk 1 above)
Comment (sternetj) [reply src/common/DeveloperConfig.ts:119]: Yeah exactly. It also has the added benefit of being able to change the default order in addition to injecting/removing default values.

**Review by markdlv — APPROVED** [review]
